src/catchup/CatchupWork.cpp

```diff
@@ -181,14 +181,15 @@ CatchupWork::doWork()
                       mCatchupConfiguration.toLedger() + 1) -
                       1;
         mGetHistoryArchiveStateWork =
-            addWork<GetHistoryArchiveStateWork>(mRemoteState, toCheckpoint);
+            addWork<GetHistoryArchiveStateWork>(toCheckpoint);
```

Comment:
Previously we were passing a reference to HAS to `GetHistoryArchiveStateWork` to load the state into (which other works, like CatchupWork, would later use), which is quite error-prone. Instead, we're now explicitly querying `GetHistoryArchiveStateWork` for the loaded state instead of relying on the reference.